businessRules/donations.js

```diff
@@ -1,54 +1,64 @@
 const { v4 } = require('uuid');
 const { createDonation } = require('../models/donations');
-const { getAllIdeas } = require('../models/ideas');
-const { getIdeas } = require('../utils/storage');
+const { validateIdea } = require('../validations/ideas');
+const { validateSum } = require('../validations/donations.js');
+const { getUser } = require('./users');
+const { getIdea } = require('./ideas');
+const { setUsers, getUsers } = require('../utils/storage');
 
-function createDonation(firstName, sum, userId, ideaId) {
-    let ideas = getIdeas();
-    const idea = ideas.find((idea) => ideaId === idea.id);
+function createAnonymousUser(firstName) {
+    let users = getUsers();
 
-    if (!idea) {
-        throw new Error('Error. No idea with such ID found.');
-    }
-
-    const generateId = v4();
-    const anonymous = 'Anonymous';
+    const userId = v4();
+    const defaultFirstName = 'Anonymous';
+    const defaultUsername = 'none';
+    const defaultPassword = 'none';
+    const defaultLastName = 'none';
+    const defaultSession = 'none';
+    const defaultRole = 'guest';
 
-    firstName = firstName || anonymous;
-    userId = userId || generateId;
+    if (!firstName) {
+        firstName = defaultFirstName;
+    }
 
-    createDonation(firstName, sum, userId, ideaId);
+    const anonymousUser = {
+        id: userId,
+        picture: './img/default_userpic.webp',
+        userName: defaultUsername,
+        password: defaultPassword,
+        firstName,
+        lastName: defaultLastName,
+        session: defaultSession,
+        role: defaultRole,
+    };
+    const newAnonymousUsers = [...users, anonymousUser];
```

Comment:
Why `newAnonymousUsers` and not `updatedUsers`?

Reply:
renamed.